cupy/cuda/cub.pyx

```diff
@@ -258,9 +267,10 @@ def device_csrmv(int n_rows, int n_cols, int nnz, ndarray values,
         n_cols, nnz, s, dtype_id)
     ws = memory.alloc(ws_size)
     ws_ptr = <void *>ws.ptr
-    cub_device_spmv(ws_ptr, ws_size, values_ptr, row_offsets_ptr,
-                    col_indices_ptr, x_ptr, y_ptr, n_rows, n_cols, nnz, s,
-                    dtype_id)
+    with nogil:
+        cub_device_spmv(ws_ptr, ws_size, values_ptr, row_offsets_ptr,
```

Comment:
Ditto. How about removing `with nogil:`?

Reply:
`cub_device_segmented_reduce` is actually a sparse Matrix-Vector multiplication method so I want to add `nogil`.